src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1651,13 +1651,20 @@ private void addCommentVisibilityToTable(
             List<FeedbackResponseAttributes> existingResponses, CourseRoster courseRoster, @Nullable String section) {
 
         // first get all possible giver recipient pairs
-        Map<String, Map<String, Set<String>>> completeGiverRecipientMap =
-                fqLogic.buildCompleteGiverRecipientMap(feedbackSession, relatedQuestionsMap.values(), courseRoster);
+        Map<String, Map<String, Set<String>>> questionCompleteGiverRecipientMap = new HashMap<>();
+        for (FeedbackQuestionAttributes feedbackQuestion : relatedQuestionsMap.values()) {
+            if (feedbackQuestion.getQuestionDetails().shouldGenerateMissingResponses(feedbackQuestion)) {
+                questionCompleteGiverRecipientMap.put(feedbackQuestion.getId(),
+                        fqLogic.buildCompleteGiverRecipientMap(feedbackSession, feedbackQuestion, courseRoster));
+            } else {
+                questionCompleteGiverRecipientMap.put(feedbackQuestion.getId(), new HashMap<>());
+            }
```

Comment:
I think the common code `questionCompleteGiverRecipientMap.put()` can be extracted out of if-else.

Reply:
Can I check if you're requesting for a change? Cause if not I was gonna merge this PR haha

Reply:
Not big issue, can merge.